packages/core/src/test/amazonqFeatureDev/session/session.test.ts

```diff
@@ -3,32 +3,110 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import * as vscode from 'vscode'
+import * as assert from 'assert'
+
 import sinon from 'sinon'
-import { createMessenger, createSession } from '../utils'
-import { Session } from '../../../amazonqFeatureDev/session/session'
+
+import { ControllerSetup, createController, createMessenger, createSession } from '../utils'
+import { CurrentWsFolders } from '../../../amazonqFeatureDev/types'
+import { CodeGenState } from '../../../amazonqFeatureDev/session/sessionState'
+import { FeatureDevClient } from '../../../amazonqFeatureDev/client/featureDev'
+import { FileSystemCommon } from '../../../srcShared/fs'
+import path from 'path'
 import { assertTelemetry } from '../../testUtil'
 
 describe('session', () => {
     const conversationID = '12345'
-
-    let session: Session
-
-    beforeEach(async () => {
-        const messenger = createMessenger()
-        session = await createSession({ messenger, conversationID })
-    })
+    const messenger = createMessenger()
```

Comment:
Non-trivial code must not be called in describe() scope. use a `before()` (not `beforeEach()`) if you want to do one-time setup.

Reply:
moved to beforeEach